libs/design-system/tooltips/rich-tooltip/src/lib/rich-tooltip.directive.ts

```diff
@@ -0,0 +1,214 @@
+import { ConnectedPosition, Overlay } from '@angular/cdk/overlay';
+import { TemplatePortal } from '@angular/cdk/portal';
+import { Directive, effect, ElementRef, inject, input, output, ViewContainerRef } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { RichTooltipContainerComponent } from './content/rich-tooltip-content.component';
+import { RichTooltipController } from './rich-tooltip.types';
+
+/**
+ * Viewport margin for the rich tooltip
+ */
+const VIEWPORT_MARGIN = 8;
+
+/**
+ * Positions array for the rich tooltip
+ */
+const POSITIONS: ConnectedPosition[] = [
+  {
+    originX: 'center',
+    originY: 'bottom',
+    overlayX: 'center',
+    overlayY: 'top',
+  },
+  {
+    originX: 'end',
+    originY: 'bottom',
+    overlayX: 'start',
+    overlayY: 'top',
+  },
+  {
+    originX: 'start',
+    originY: 'bottom',
+    overlayX: 'end',
+    overlayY: 'top',
+  },
+  {
+    originX: 'center',
+    originY: 'top',
+    overlayX: 'center',
+    overlayY: 'bottom',
+  },
+  {
+    originX: 'end',
+    originY: 'top',
+    overlayX: 'start',
+    overlayY: 'bottom',
+  },
+  {
+    originX: 'start',
+    originY: 'top',
+    overlayX: 'end',
+    overlayY: 'bottom',
+  },
+];
+
+/**
+ * HRA Rich Tooltip Directive
+ */
+@Directive({
+  selector: '[hraRichTooltip]',
+  host: {
+    '(click)': 'toggle()',
+  },
+})
+export class RichTooltipDirective implements RichTooltipController {
+  /**
+   * Custom content for the rich tooltip
+   * - Not required if need to use simple content variant
+   */
+  readonly customContainer = input<RichTooltipContainerComponent>(undefined, { alias: 'hraRichTooltip' });
+
+  /**
+   * Title for the rich tooltip
+   * - Not required if using custom content variant
+   */
+  readonly tagline = input<string>(undefined, { alias: 'hraRichTooltipTagline' });
```

Comment:
Let's update the default value to the empty string `''` so it doesn't show as the string `'undefined'` when it is used in a custom content without a title element